File: src/lira/linguistics/data/linguistic_unit_kind.py
from enum import Enum


class LinguisticUnitKind(Enum):
    """Values are the numeric codes this kind is stored as in
    LinguisticSystemPropertyTensor (Rule 14) -- .value is used directly
    as the tensor cell, not just a label.

    Word and Punctuation are both Vocabulary's Word class at the Python
    type level (a punctuation mark is a Word with
    part_of_speech=PartOfSpeech.PUNCTUATION, not a separate class) --
    GraphProcessor.process_token still tags a token's tensor row with
    the Punctuation kind rather than Word when that's the case, derived
    from part_of_speech instead of an isinstance check."""

    Word = 0
    Punctuation = 1
    Clause = 2
    Sentence = 3
    Paragraph = 4
    Subject = 5
    UserPrompt = 6

File: prototype/src/lira/linguistics/data/linguistic_unit_kind.ts
/** Values are the numeric codes this kind is stored as in
 * LinguisticSystemPropertyTensor (Rule 14) -- the numeric value is used
 * directly as the tensor cell, not just a label.
 *
 * Word and Punctuation are both Vocabulary's Word class at the type
 * level (a punctuation mark is a Word with partOfSpeech=PUNCTUATION,
 * not a separate class) -- GraphProcessor.materialiseToken still tags a
 * token's tensor row with the Punctuation kind rather than Word when
 * that's the case, derived from partOfSpeech instead of an instanceof
 * check.
 *
 * Ported from linguistics/data/linguistic_unit_kind.py. */
export enum LinguisticUnitKind {
  Word = 0,
  Punctuation = 1,
  Clause = 2,
  Sentence = 3,
  Paragraph = 4,
  // Renamed from Subject (prototype only -- see data/document.ts's own
  // docstring); same tensor slot, same underlying concept, so the
  // value stays 5 -- a rename is not a renumbering.
  Document = 5,
  UserPrompt = 6,
  // Appended, not inserted -- 0-6 above are already live tensor cell
  // codes in any LinguisticSystemPropertyTensor built before this value
  // existed; renumbering them would silently corrupt previously-
  // allocated rows' kindOf() reads.
  Phrase = 7,
  // Also appended, same reasoning -- new to this hierarchy (data/heading.ts).
  Heading = 8,
}
